src/key.rs

```diff
@@ -1053,13 +1053,18 @@ impl<'a> From<&'a KeyPair> for PublicKey {
     }
 }
 
+#[cfg(feature = "alloc")]
```

Comment:
The `alloc` requirement due to calling `Secp256k1::signing_only()` is a bit annoying, but better than the `global-context` one

Reply:
I wonder if we should have a non-`alloc` non-`global-context` version that just returns an error 100% of the time.

My goal here is really just that we don't break compilation for anybody, while replacing the current "always aborts the process" code with something else...I don't care too much **what** else since the current behavior is so bad :)

Reply:
The code has gotten a bit hairy, but I think it should have the desired behavior now.